### DISCUSSION TITLE
fail to build lammps-dp

### DISCUSSION THREAD

**Discussion by imindshejian** [deepmodeling/deepmd-kit#571] [Q&A]
env:  Ubuntu 20.04 2080Ti gcc-4.8.5 cmake 3.18.5  bazel-0.25.1 cuda-10.1 cudnn-7.6 miniconda 4.9.2
I succesfully build tensorflow C++  and deepmd-kit C++ .
when I try to build lammps-dp

cd lammps/src
cp -r ../../deepmd-kit/source/build/USER-DEEPMD/ .
make yes-user-deepmd
make mpi 

then a error message :

make[1]: *** [Makefile:109：pppm_dplr.o] error 1
make[1]: ***....
In file included from ../fix_dplr.h:12:0,
                 from ../style_fix.h:15,
                 from ../modify.cpp:16:
**.**./pair_nnp.h:24:22: fatal error: NNPInter.h: No such file or dictionary****
 #include "NNPInter.h"
                      ^
compilation terminated.
make[1]: *** [Makefile:109：force.o] error 1
In file included from ../fix_dplr.h:12:0,
                 from ../fix_dplr.cpp:13:
../pair_nnp.h:24:22: fatal error: NNPInter.h: No such file or dictionary
 #include "NNPInter.h"
                      ^
compilation terminated.
make[1]: *** [Makefile:109：lammps.o] error 1
make[1]: *** [Makefile:109：modify.o] error 1
make[1]: *** [Makefile:109：fix_dplr.o] error 1

what make this error ?

**Comment by njzjz** [comment]
Please check your `Makefile.package` file under src directory and see whether the path is correct.

**Reply by imindshejian** [reply]
i checked Makefile.package,  found that /home/kemove/tf/tensorflowcc/include/deepmd was included, but deepmd was not in the tensorflowcc/include , so i copied deepmdcc/include/deepmd to this path, this error didn't show agian.
but another error occurred,  **ld:  -ldeepmd_op_cuda  -ldeepmd_op  -ldeepmd** not found  , it is writted in Makefile.package line 6 : **PKG_LIB =     -Wl,--no-as-needed -ldeepmd_op_cuda -ldeepmd_op -ldeepmd**  what are these path ?

**Reply by njzjz** [reply]
You need to check the directory after `-L`. It should include the directory that contain `libdeepmd_op_cuda.so` and so on. See gcc document: https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html
> The linker searches a standard list of directories for the library. The directories searched include several standard system directories plus any that you specify with -L.

**Reply by imindshejian** [reply]
thank you,
